StShared/StResourceManager.cpp

```diff
@@ -162,20 +162,35 @@ StResourceManager::StResourceManager(const StString& theAppName)
     StFolder::createFolder(myUserHomeFolder + "Library/Caches");
 #else
     // Linux world
-    myUserDataFolder = myUserHomeFolder + ".local/share/" + myAppName + "/";
-    mySettingsFolder = myUserHomeFolder + ".config/"      + myAppName + "/";
-    myCacheFolder    = myUserHomeFolder + ".cache/"       + myAppName + "/";
+    myUserDataFolder = StProcess::getEnv(StString("XDG_DATA_HOME"));
+    if (myUserDataFolder.isEmpty()) {
+        myUserDataFolder = myUserHomeFolder + ".local/share/" + myAppName + "/";
+        StFolder::createFolder(myUserHomeFolder + ".local");
+        StFolder::createFolder(myUserHomeFolder + ".local/share");
+    } else {
+        myUserDataFolder = myUserDataFolder + "/";
+    }
+
+    mySettingsFolder = StProcess::getEnv(StString("XDG_CONFIG_HOME"));
```

Comment:
These XDG variables define location of user folders, not dedicated to current application.
So that to keep logic, myAppName should be appended.

Reply:
But they're really only ever set by Flatpak, which already makes those directories per-app. But sure.